### ISSUE TITLE
Make zenity, kdialog, qarma as a dependency

### ISSUE THREAD

**Issue by sunnykarira-ms** [miguelpruivo/flutter_file_picker#1451]
Similar Issue: https://github.com/miguelpruivo/flutter_file_picker/issues/1270

**Describe the bug**
Since the dependency for zenity, kdialog, qarma is not mentioned anywhere in the package. Would it be fine to add libraries as a dependency or a mention that this package depends on the above one.

**Platform**

- [ ] Android
- [ ] iOS
- [ ] Web
- [X] Desktop

**Platform OS version**
What version did it happen?
Ubuntu 20, Ubuntu 22

**How are you picking?**
```
 var result = await FilePicker.platform.pickFiles(
        type: FileType.custom,
        allowedExtensions: ["xml"],
      );
```

**Details to reproduce the issue**

**Error Log**
1. Install on an Ubuntu 22 machine which does not have zenity installed. Launch the app and try to import a profile. The import button does nothing, and no error message is reported to the user.

If you launch the app from the command line, you will see these logs:

[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: Exception: Couldn't find the executable zenity in the path.
#0      isExecutableOnPath (package:file_picker/src/utils.dart:60)
<asynchronous suspension>
#1      FilePickerLinux._getPathToExecutable (package:file_picker/src/linux/file_picker_linux.dart:120)
<asynchronous suspension>
#2      FilePickerLinux.pickFiles (package:file_picker/src/linux/file_picker_linux.dart:23)
<asynchronous suspension>
#3      Utilities.importProfile (package:/utils.dart:74)
<asynchronous suspension>
#4      _ImportProfile.build.<anonymous closure> (package:/import.dart:31)
<asynchronous suspension>
2. Install "zenity" using apt. "sudo apt install -y zenity". Now the import functionality works as expected.

Linux has been standardizing some features recently and there's a File Chooser XDG Portal which can be used with native integration with the user system as well as working very well inside sandboxes (since they were created to work in them).
Should we use XDG portal internally?

*Expected Behavior:*
User should not be expected to identify and install dependencies.

**Comment by miguelpruivo** [comment]
@sunnykarira-ms since I'm not the main maintainer of Desktop side and it's typically @philenius, what would you suggest to add to the docs in order to fix this? I'm all hears, let me know.

Thanks!

**Comment by gjwgit** [comment]
In our  Rattle project using file_picker we came across this issue, and more, on Ubuntu 24.04. We've documented it as https://github.com/gjwgit/rattleng/issues/395.

In short, on Linux, file_picker uses an external app designed for script-based file picking. It attempts to find and so use qarma, kdialog, zenity, in that order. 

Zenity is usually found installed by default on Ubunut/Gnome, but none are found on Linux Mint 21.1 (and hence an exception on those systems when a file is to be picked).

I have found issues with Zenity, including not always populating the default filename. See https://github.com/gjwgit/rattleng/issues/395.

Current conclusion for any Linux users of your desktop app developed in Flutter using file_picker, **make sure kdialog** is installed, and remove qarma if it is installed.
